### PR TITLE
Feature/standard census circuit

### PR DESCRIPTION
**Pull request by lucasmenendez**
New circuit that allows to a voter to proof that is part of weighted census through zkSnarks, based on voter private key and voter weight. This is intended to standardize the way which the census proof is verified between weighted census with and without zkSnarks.

### CODE REVIEW

**Review by p4u — APPROVED** [review]
LGTM

File: circuits/census.circom [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
-                       | Verifier |            |          |
-PRI_siblings+--------->+          |(value)<----+ Poseidon +<-----+--+PRI_secretKey
-                       |          |            |          |      |
-                       +----------+            +----------+      |
-                                                                 |
-                                     +----------+                |
-                      +----+         |          +<---------------+
+					   +----------+
+					   |          |
+PUB_censusRoot+------->+          |(value)<-----+PUB_weight
```
Comment (arnaucube) [inline circuits/census.circom]: seems to be displaced now: https://github.com/vocdoni/zk-franchise-proof-circuit/blob/feature/standard-census-circuit/circuits/census.circom#L16

**Review by arnaucube — COMMENTED** [review]
Nice!! :raised_hands:

**Comment by arnaucube** [comment]
Related to the approach, initially we were using the *secretKey* (now *privateKey*) as input of the circuit in order to prove its ownership while keeping the number of constraints low (it is cheaper to hash the secretkey than to verify a BBJJ signature), but given that the signature verification is about ~4k constraints, it does not add much extra computation (even in the browser) and maybe it makes the flow more compatible with future designs.

As an example, in [Ovote different user votes are aggregated](https://github.com/aragonzkresearch/ovote/blob/main/circuits/src/ovote.circom) by proving inside the circuit the validity of each signature, but then, in the [OAV circuit the same signature can be used](https://github.com/aragonzkresearch/ovote/blob/main/circuits/src/oav.circom) to generate a zkproof by the user proving their membership to the census.

It is not a blocker, as the user could use the no-signature-approach for the anonymous voting case and the signature-approach (even with other curves such as eth curve) for the other cases but maybe this makes the flow more generic, and even if it is needed in the future the voter could send the signature to the Vochain for those cases where the zkproof is not needed.

For example, there could be 3 different flows with the same voter babyjubjub signature:
the voter always sings their vote, and depending on the case:
- case onchain non-anonymous: user sends their signature of the vote to the Vochain (no zkproof, with the rest of stuff such as merkleproof, etc)
- case onchain anonymous: user sends a zkproof proving the validity of their signature (without revealing the sig, and proving the rest of stuff)
- case offchain aggregated: user sends their signature (with the rest of stuff such as merkleproof, etc) to a "Aggregator" which will generate a proof of correct verification of all the voter's signatures

File: circuits/census.circom (hunk 1 above)
Comment (lucasmenendez) [reply circuits/census.circom]: solved!

**Pull request merged**